File: java/sum-of-multiples/src/main/java/SumOfMultiples.java
class SumOfMultiples {

    private int number;
    private int[] set;

    SumOfMultiples(int number, int[] set) {
        this.number = number;
        this.set = set;
    }

    int getSum() {
        int sum = 0;
        for (int i = 1; i < number; i++) {
            for (int s: set) {
                if (i % s == 0) {
                    sum += i;
                    break;
                }
            }
        }
        return sum;
    }

}

File: cpp/sum-of-multiples/sum_of_multiples.cpp
#include "sum_of_multiples.h"

namespace sum_of_multiples {

    int to(const std::set<int> &integers, int number) {
        int sum = 0;
        for (int i = 1; i < number; i++) {
            for (const auto &j : integers) {
                if (i % j == 0) {
                    sum += i;
                    break;
                }
            }
        }
        return sum;
    }

}
